test/NuGet.Core.Tests/NuGet.DependencyResolver.Core.Tests/RemoteDependencyWalkerTests.cs

```diff
@@ -1572,8 +1572,8 @@ public async Task CentralTransitiveDependency_NoDowngrades_IfTwoDowngrades_WereR
 
         /// <summary>
         /// A -> B 1.0.0 -> C 1.0.0(this will be rejected) -> D 2.0.0(this will be downgraded due to central 1.0.0) -> E 1.0.0
-        ///   -> F 1.0.0 -> C 2.0.0 -> H 2.0.0
-        ///   -> G 1.0.0 -> H 2.0.0(this will not be rejected) -> D 1.0.0
+        ///   -> F 1.0.0 -> C 2.0.0
+        ///   -> G 1.0.0 -> H 2.0.0(this will not be rejected) -> D 1.0.0 (this will be downgraded due to central 1.0.0)
```

Comment:
Does `H 2.0.0(this will not be rejected)` reflect current new setup? You removed `-> H 2.0.0` from above line, so there wouldn't be any rejection concern for `H`.

Reply:
Version of `D` doesn't match on line 1607.

Reply:
The test was asserting an incorrect behavior. It was asserting a downgrade from a pruned codepath, which should not be happening. 

This is the only inconsistency between setup and description I found..

Reply:
Yes, it just setup and desc inconsistency. I just following unit test to understand more.